Formula/rpm.rb

```diff
@@ -32,10 +36,18 @@ class Rpm < Formula
   depends_on "xz"
   depends_on "zstd"
 
-  # Fix -flat_namespace being used on Big Sur and later.
+  # Fix `fstat64` detection for Apple Silicon.
+  if Hardware::CPU.arm?
```

Comment:
Upstream need a different patch to make this change work on Intel macOS too. Will look into it tomorrow.